lib/igniter.ex

```diff
@@ -504,21 +504,51 @@ defmodule Igniter do
   """
   @spec update_elixir_file(t(), Path.t(), zipper_updater()) :: Igniter.t()
   def update_elixir_file(igniter, path, func) do
-    if Rewrite.has_source?(igniter.rewrite, path) do
-      igniter
-      |> apply_func_with_zipper(path, func)
-      |> format(path)
-    else
-      if exists?(igniter, path) do
-        source = read_ex_source!(igniter, path)
-
-        %{igniter | rewrite: Rewrite.put!(igniter.rewrite, source)}
-        |> format(path)
+    case use_or_read_elixir_file(igniter, path) do
+      {:exists, igniter} ->
+        igniter
         |> apply_func_with_zipper(path, func)
         |> format(path)
-      else
+
+      {:missing, igniter} ->
         add_issue(igniter, "Required #{path} but it did not exist")
-      end
+    end
+  end
+
+  @doc """
+  Updates the source code of the given elixir file if it exists
+  """
+  @spec update_existing_elixir_file(t(), Path.t(), zipper_updater(), keyword) :: Igniter.t()
+  def update_existing_elixir_file(igniter, path, func, opts \\ []) do
```

Comment:
could this instead be done via `update_elixir_file(...., required?: false)`? Then we wouldn't need another function.

Reply:
Probably. I didn't find the other related functions to be particularly consistent around this topic.

Reply:
What we have currently is an `on_exists` configuration for `create_new_file` for example, so perhaps we can have an `on_not_exists` configuration which defaults to `:error`.

Reply:
There's also a `required?` option for `include_existing_file` with an `include_or_create_file` counterpart. I'm not particularly interested in any option over the other. I don't think there's much of a line right now though.